### DISCUSSION TITLE
Android Support?

### DISCUSSION THREAD

**Discussion by mwaitzman** [espanso/espanso#1252] [Feature requests and Ideas]
What's the current (and planned) state of using Espanso on Android?

**Comment by federico-terzi** [comment]
Hey @mwaitzman,

Thanks for your interest! At the moment, the Android version is only a long-term vision, as there are other more important features we'll need to tackle first (GUI, Macros, and a few others). It's unlikely for the Android version to come before 2023

I'll make sure to update the issue when something new happens in this area!

Cheers :)

**Reply by beedell-roke** [reply]
> https://github.com/espanso/espanso/discussions/1252#discussioncomment-3069212

@federico-terzi, please reopen https://github.com/espanso/espanso/issues/939 so that I can subscribe solely to issue closure.

**Comment by usernotnull** [comment]
@federico-terzi I've been using espanso for a year now and I'm incredibly grateful for your contribution!
Any progress on the Android version of this app?

**Comment by AaronKnowls** [comment]
I'm interested in this too.

Thanks for this brilliant software!

**Comment by dacog** [comment]
Hi @federico-terzi 

I just made a first version of an Android App that uses `espanso` match YAML files to trigger replacements, and I thought it would be interesting for this discussion. You can just upload the `espanso` folder to your phone (or sync it with another app) and you are ready-to-go. 

The app uses the accessibility service in Android to expand the texts, and it can also browse the triggers and their replacements in the app's UI and copy them to the clipboard. It is under the same license as `espanso`

## Features

- **Parse YAML Files:** Reads `.yml` files from the `Documents/espanso/match` directory to load text expansion triggers and values.
- **List of Files and Triggers:** Displays the list of YAML files and their triggers and values.
- **Search Functionality:** Provides a search bar (in-app) to filter triggers and values.
- **Clipboard Copy:** Copies the value of a trigger to the clipboard when a trigger-value pair is clicked.
- **Text Expansion:** Expands text shortcuts into longer predefined phrases in any app using an Accessibility Service.

## Download

You can check the app here:

https://github.com/dacog/textexpander_android/

and download it here:

https://github.com/dacog/textexpander_android/releases/tag/1.0

**Reply by federico-terzi** [reply]
@dacog That's amazing! Feel free to share it on Reddit and Discord, I'm sure users will be super happy :) Also, if you want to work more on it, we could also make it part of the Espanso organization as the official Espanso for Android and you'd be the official maintainer. That said, if you want to just keep it as fun exploration it's also ok, and really impressive!

**Reply by dacog** [reply]
Hi @federico-terzi. Thank you for your feedback! I already posted it on [Discord](https://discordapp.com/channels/884163483409731584/1013915939156275210/1246927287488741416) and [Reddit](https://www.reddit.com/r/espanso/comments/1d67tnt/android_app_compatible_with_espanso_version_1/).

Making my app the official Espanso version for Android sounds great. I think it would really help more people in the community start using it.

I have a couple of questions:

What would it mean to me and the app to make it part of Espanso? 
For example,
 -  make use of the logo, colors and so on to make the app look more like espanso, 
 - using the namespace of espanso,  
 - adding it to the official website,
 - and so on...

And what would be expected from my side?
- My main focus now would be to develop features that I need and update the app so that it works in new Android versions, and while I’m eager to contribute as much as I can, my time is a bit limited.
- What do you and the espanso contributors expect from an official maintainer, mainly in terms of time and responsibilities?

Please share your thoughts on this. Should we continue discussing this here, or would you prefer to move our conversation to Discord?

**Comment by smeech** [comment]
That looks very interesting. Sadly, my ancient Android devices aren't up to spec to try it, but I look forward to hearing from my colleagues.

May I suggest you also announce it on the [Discord](https://www.reddit.com/r/espanso/comments/184fxch/official_discord_channel/) server, as that's where most day-to-day discussion takes place? Your project might find a place in our Collaboration channel, and an entry in the docs [Community Tools & Plugins](https://espanso.org/docs/tools/) in due course.

**Reply by dacog** [reply]
Hi @smeech ! Thank you for your answer! I will take a look at your links :)

Which android versions do you have? I may take a look at them using the emulators when I get some time again.  The app should support Android from api version 22 (android 5.0) to version 34 (android 14), according to the gradle config.

**Reply by smeech** [reply]
My devices are both Android 8. Your prerequisites refer to a device or emulator running Android 13 or higher.

**Reply by dacog** [reply]
Hi @smeech 

The requirements were there only because it was what I tested. I just tested on an emulator running API 25 (Android 7) and it worked just fine, so I have updated the README accordingly. 

Feel free to try it on your devices. :)

**Comment by diegomarzaa** [comment]
Hi, sharing another open-source option for Espanso users on Android.

**Expanda** (https://github.com/diegomarzaa/expanda) reads the same match YAML as desktop Espanso. You can link your `match/` folder (Syncthing, Drive, local copy) and edit snippets on phone or PC.

- License: GPLv3
- No Internet permission / no accounts
- APK: GitHub Releases + Obtainium

I'm not an Android developer by trade; the project grew with AI assistance and I'd love help from people who know accessibility services, Kotlin/Compose, or Espanso internals. Good entry points: F-Droid packaging (#2), locale-aware date UI (#3), and general review of the expansion engine.

Happy to answer compatibility questions or compare YAML behavior with desktop Espanso.
